File: BasicOperators/src/main/java/reducing_operators/count/Launcher.java
package reducing_operators.count;

import io.reactivex.Observable;

public class Launcher {

    public static void main(String[] args) {
        Observable.just("Alpha", "Beta", "Gamma", "Delta", "Epsilon")
                .count()
                .subscribe(s -> System.out.println("Received: " + s));
    }

}

File: BasicOperators/src/main/java/basic_operators/Launcher.kt
import io.reactivex.Observable

fun main() {
    val myStrings: Observable<String> = Observable.just("Alpha", "Beta", "Gamma", "Delta", "Epsilon")
    myStrings
            .doOnTerminate {
                println("do Onterminate")
            }

            .doOnNext {
                println("doOnnext: $it")
            }
            .doOnComplete {
                println("doOncomplet")
            }
            .map {
                it.length
            }
            .map {
                it.plus(it).equals(10)
            }
            .subscribe {
                println(it)
            }
}

